docs/prune.md

```diff
@@ -0,0 +1,126 @@
+# prune
+
+```
+cli-utils prune <dir>
+```
+can prune obsolete resources from a kustomization directory or resources that are with declarative deletion annotation.
+
+## prune kustomize directory
```

Comment:
Introduce that there are 2 methods for deleting resources:

- prune by lifecycle annotations (preferred)
  - explicit deletion of resources by annotating them with deletion
  - will not delete generated secrets and configmaps
- prune by inventory configmap
  - implicit deletion of resources by removing the resource from the configuration
  - will delete generated secrets and configmaps

Reply:
prune by inventory configmap doesn't delete generated secrets and configmaps if they are/were refereed in other objects.

Reply:
updated